activerecord/lib/active_record/enum.rb

```diff
@@ -60,7 +60,7 @@ module ActiveRecord
   # database integer with a hash:
   #
   #   class Conversation < ActiveRecord::Base
-  #     enum status: { active: 0, archived: 1 }
+  #     enum :status, { active: 0, archived: 1 }
```

Comment:
```suggestion
  #     enum :status, active: 0, archived: 1
```

We should be able to remove the Hash braces here.

Reply:
Unfortunately, it is hard thing to ommit the Hash braces for the second argument.

If `enum :status, default: 0, active: 1, archived: 2`, `default: 0` is ambiguous whether an enum label or `:default` option.

https://github.com/rails/rails/blob/02ad7ba5ef3480364c5dd015924fa644c8704e1c/activerecord/lib/active_record/enum.rb#L161-L162

Reply:
Sure, I could see that. But should we define `default: nil` etc. as keyword arguments then?

Reply:
I've just implemented it in ea86dce9c4a8d3f2e45730e3edda910a2153443e.
But I'm not sure that kwargs includes both enum values and options is whether pretty or not though...

https://github.com/rails/rails/blob/ea86dce9c4a8d3f2e45730e3edda910a2153443e/activerecord/test/cases/enum_test.rb#L692-L697

Reply:
I'd probably want these syntaxes to be supported:

```ruby
enum :cover, %i[ hard soft ], suffix: true
enum :cover, hard: 0, soft: 1
enum :cover, { hard: 0, soft: 1 }, suffix: true 
```

If we can't have `enum :cover, hard: 0, soft: 1` then that's ok. With ea86dce can we still have the `enum :cover, %i[ hard soft ], suffix: true` syntax, it's not clear to me?

Reply:
With ea86dce9c4a8d3f2e45730e3edda910a2153443e it still support these all syntaxes.

```ruby
enum :cover, %i[ hard soft ], suffix: true
enum :cover, hard: 0, soft: 1, suffix: true
enum :cover, { hard: 0, soft: 1 }, suffix: true
```

My concern was that people may be confused to use reserved options if the doc officially recommends to ommit the Hash braces.

Anyway if you'd prefer the syntax then I'll apply the suggestion for the doc.

Reply:
Or, one more option is that, allow `enum :cover, hard: 0, soft: 1` but don't allow `enum :cover, hard: 0, soft: 1, suffix: true`.

Reply:
@kamipo yeah, I like that. So you'd have to pass `enum :cover, { hard: 0, soft: 1 }, suffix: true`, right?

Reply:
I've implemented the idea https://github.com/rails/rails/pull/41328#discussion_r576163415 in 3e8bf3c01e9dc82d7119e4511a069aa3a70e195b.

My concern would be addressed by the change.

https://github.com/rails/rails/blob/3e8bf3c01e9dc82d7119e4511a069aa3a70e195b/activerecord/test/cases/enum_test.rb#L704-L716

Reply:
> So you'd have to pass `enum :cover, { hard: 0, soft: 1 }, suffix: true`, right?

Right, IMHO `enum :cover, hard: 0, soft: 1, suffix: true` is a bit confusing for readers.

Reply:
Cool, sounds good 👍